servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -335,6 +337,56 @@ public EntriesResponse getEntries(String namedRef, EntriesParams params)
     }
   }
 
+  @Override
+  public EntriesResponse getNamespaceEntries(
+      String refName,
+      @Nullable String hashOnRef,
+      @Nullable ContentsKey namespacePrefix,
+      @Nullable Integer depth)
+      throws NessieNotFoundException {
+    WithHash<NamedRef> refWithHash = namedRefWithHashOrThrow(refName, hashOnRef);
+    try {
+      List<EntriesResponse.Entry> entries;
+      try (Stream<EntriesResponse.Entry> keyStream =
+          getStore()
+              .getKeys(refWithHash.getHash())
+              .map(
+                  key ->
+                      EntriesResponse.Entry.builder()
+                          .name(fromKey(key.getValue()))
+                          .type((Type) key.getType())
+                          .build())) {
+        entries =
+            filterEntries(
+                    keyStream,
+                    (namespacePrefix == null || namespacePrefix.getElements().isEmpty())
+                        ? null
+                        : buildRegex(namespacePrefix))
+                .map(e -> truncate(e, depth))
+                .distinct()
+                .collect(ImmutableList.toImmutableList());
+      }
+      return EntriesResponse.builder().addAllEntries(entries).build();
+    } catch (ReferenceNotFoundException e) {
+      throw new NessieNotFoundException(
+          String.format("Unable to find the reference [%s].", refName), e);
+    }
+  }
+
+  private String buildRegex(ContentsKey namespacePrefix) {
+    return String.format(
+        "entry.namespace.matches('%s(\\\\.|$)')",
+        Joiner.on("\\\\.").join(namespacePrefix.getElements()));
```

Comment:
Should we also regex-quote each element. Are names allowed to contain special chars?

Reply:
Would it be more robust to have a custom prefix-matching filter in java?